From 0f5930b195f0fc25fd47e1aa6b7fe67839b13e73 Mon Sep 17 00:00:00 2001
From: yannick Fertre <yannick.fertre@st.com>
Date: Fri, 13 Feb 2015 11:57:54 +0100
Subject: [PATCH 10/15] libv4l-hva: manage frame packing arrangement

Need to send controls of frame packing arrangement, require by profile stereo or multiview.

Change-Id: Ida41dbab207dda67cdcc67132ed8ee98a6017d30
Signed-off-by: yannick Fertre <yannick.fertre@st.com>
Reviewed-on: https://gerrit.st.com/23271
Reviewed-by: Karine BOCLAUD <karine.boclaud-ext@st.com>
---
 lib/libv4l-hva/libv4l-hva-h264.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/lib/libv4l-hva/libv4l-hva-h264.c b/lib/libv4l-hva/libv4l-hva-h264.c
index 1a8c9db..b6ff65c 100644
--- a/lib/libv4l-hva/libv4l-hva-h264.c
+++ b/lib/libv4l-hva/libv4l-hva-h264.c
@@ -31,6 +31,8 @@
  * @format: stream format
  * @sar_idc: sample aspect ratio index
  * @sar_enable: enable sample aspect ratio
+ * @sei_fp_type: frame packing arrangement type
+ * @sei_fp_enable: enable frame packing arrangement
  */
 struct hva_h264_meta {
 	/* controls */
@@ -49,6 +51,9 @@ struct hva_h264_meta {
 	struct v4l2_format format;
 	enum v4l2_mpeg_video_h264_vui_sar_idc sar_idc;
 	int sar_enable;
+	/* Supplemental Enhancement Information */
+	enum v4l2_mpeg_video_h264_sei_fp_arrangement_type sei_fp_type;
+	int sei_fp_enable;
 };
 
 static const GstVaapiProfile
@@ -186,6 +191,16 @@ hva_h264_set_ctrl(void *priv, struct v4l2_ext_controls *ctrls)
 			V4L2_LOG("V4L2_CID_MPEG_VIDEO_H264_VUI_SAR_ENABLE %d\n",
 				ctrls->controls[i].value);
 		break;
+		case V4L2_CID_MPEG_VIDEO_H264_SEI_FRAME_PACKING:
+			meta->sei_fp_enable = ctrls->controls[i].value;
+			V4L2_LOG("V4L2_CID_MPEG_VIDEO_H264_SEI_FRAME_PACKING %d\n",
+				ctrls->controls[i].value);
+		break;
+		case V4L2_CID_MPEG_VIDEO_H264_SEI_FP_ARRANGEMENT_TYPE:
+			meta->sei_fp_type = ctrls->controls[i].value;
+			V4L2_LOG("V4L2_CID_MPEG_VIDEO_H264_SEI_FP_ARRANGEMENT_TYPE %d\n",
+				ctrls->controls[i].value);
+		break;
 		default:
 			V4L2_LOG_ERR("VIDIOC_S_EXT_CTRLS(): Unsupported control id %d\n",
 				ctrls->controls[i].id);
-- 
2.7.4

